compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -12849,25 +12849,24 @@ TR::Register *OMR::Z::TreeEvaluator::iRegStoreEvaluator(TR::Node *node, TR::Code
     }
 
     if (!useLGHI) {
-        globalReg = needsLGFR && noLGFgenerated ? cg->gprClobberEvaluate(value) : cg->evaluate(value);
+        // Force the sign extension on the register associated with value so that
+        // when encountering the value under the i2l, we can skip the conversion.
+        globalReg = cg->evaluate(value);
```

Comment:
I'm having some difficulty understanding this comment.  How is the call to `evaluate` forcing the sign extension to happen?

Reply:
Value under `iRegStore` is evaluated and signExtended if regStore needs sign extension. It is not the evaluate that forces the sign extension, but the flag being set after generating sign extension instruction forces the sign extension. 

Removing the comment, as I think it is more confusing.